### DISCUSSION TITLE
Inconsistent cursor size with some xwayland apps

### DISCUSSION THREAD

**Discussion by TheCheddarCheese** [hyprwm/Hyprland#7842] [Q&A]
When I use some xwayland apps, the cursor becomes really big. It seems to happen with programs not made with a UI toolkit (like minecraft or feh) and some gtk apps (timeshift & gparted). I have a theme installed for both hyprcursor and xcursor, both set to size 24 (though it happens regardless of what xcursor size i set) and fractional scaling turned off

![20240918_23h08m29s_grim](https://github.com/user-attachments/assets/3ddee4f4-cc16-4e38-886b-6bd796246e5d)

**Comment by ghost** [comment]
Have you checked your gsettings or gtk settings.ini files to make sure your cursor is set correctly for gtk apps? Do you have environment variables set for GTK or QT themes?

**Reply by TheCheddarCheese** [reply]
I checked, the size is set correctly. And I don't have any variables set aside from the ones I showed above

**Comment by TheCheddarCheese** [comment]
What's weird is that it doesn't happen with the adwaita cursor. If anyone has any info about this please let me know, it's really bugging me

**Comment by TheCheddarCheese** [comment]
I looked around the arch wiki and fixed the issue by changing the cursor theme to rosé pine and editing `/usr/share/icons/default/index.theme` to include it (so that it doesn't change back to adwaita). So the problem was with the Nordzy theme specifically, although I'm not sure what exactly caused it to happen.

**Reply by CyrenArkade** [reply]
The problem is that different xcursor sizes are used, despite them appearing the same size on screen. If you use a theme that doesn't have _both_ of the sizes, the scaling will be applied despite falling back to a larger size that shouldn't be scaled. You can see this in the below screenshots where Hyprland is using the 48px cursor despite XCURSOR_SIZE being set to 24.
<img width="177" height="200" alt="2025-07-20_01:05:41" src="https://github.com/user-attachments/assets/fe7a8933-d167-4862-80e7-248b19c29c86" />
<img width="85" height="67" alt="2025-07-20_01:05:01" src="https://github.com/user-attachments/assets/0c25f010-0f94-430b-b874-1c124f5fba21" />
<img width="108" height="84" alt="2025-07-20_01:04:47" src="https://github.com/user-attachments/assets/4d6a399b-f2b4-42cb-97ee-b48980ea865c" />
